mesonbuild/mparser.py

```diff
@@ -266,11 +266,19 @@ def accept(self, visitor: 'AstVisitor') -> None:
             if callable(func):
                 func(self)
 
+@dataclass(init=False)
```

Comment:
init=False is not strictly needed since these classes have their own init methods; dataclasses is documented to ignore the value of init= and refrain from overwriting an existing method.

Reply:
I gave that a try, but it's needed otherwise I get `TypeError: non-default argument 'value' follows default argument`. That's because BaseNode has default value for end_lineno and end_colno. This is arguably a python bug.

I would suggest that fix: https://github.com/mesonbuild/meson/commit/2f05c20a74074ff2080c83d9b21c7fd2d3f3e918